test/e2e/e2e_test.go

```diff
@@ -690,6 +683,88 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQueryForVector(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (model.Vector, error) {
+	results, warnings, err := promClient.Query(context.Background(), query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
+		return model.Vector{}, err
+	}
+	if len(warnings) > 0 {
+		t.Logf("prometheus query emitted warnings: %v", warnings)
+	}
+
+	result, ok := results.(model.Vector)
+	if !ok {
+		t.Fatalf("expecting scalar type result, found: %v ", reflect.TypeOf(results))
+	}
+
+	return result, nil
+}
+
+func getSampleForPromQuery(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (*model.Sample, error) {
+	res, err := runPromQueryForVector(t, promClient, query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
```

Comment:
nit: Given that `runPromQueryForVector` returns formatted errors, `t.Errorf(err)` is likely sufficient.  But again, is logging necessary if the error is returned?

Reply:
Same! handled as i said in my previous comments.